### PR TITLE
Nisyad/add detection confusion matrix

### PR DESCRIPTION
**Pull request by nisyad-ms**

### CODE REVIEW

File: visionmetrics/detection/confusion_matrix.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                            best_gt_index = gt_index
+
+                if best_gt_index != -1:
+                    if best_iou >= iou_threshold:
+                        self.tp += 1
+                        gt_used[best_gt_index] = True
+                    else:
+
+                        self.fp += 1
+                        self.fp_due_to_low_iou += 1
```
Comment (cy-bai) [inline visionmetrics/detection/confusion_matrix.py:95]: I think this still includes errors that are more like wrong class to me. For example, ground truth box [0,0,10,10] label 0, [90,90,100,100] label 1, prediction [90,90,100,100] label 0.  In current prediction this belongs to "fp_due_to_wrong_class". How about for boxes that don't overlap w/ any gt, they belong to "no corresponding gt box", for boxes that have overlap but labels are wrong, they belong to "wrong class"?

Comment (nisyad-ms) [reply visionmetrics/detection/confusion_matrix.py:95]: I had added a note in the definition that when both "fp_due_to_wrong_class" and "fp_due_to_low_iou" conditions are satisfied, "fp_due_to_wrong_class" will be given precedence. . For example, if two boxes don't overlap and have wrong classes then the code assigns it as "fp_due_to_wrong_class" but if in the case of two non-overlapping boxes have same class they get assigned to "fp_due_to_low_iou". 

We could also flip the precedence order. I don't have a strong take on it. In that case, any two boxes with low iou (regardless of class ids) get assigned to "fp_due_to_low_iou". And FPs get assigned to "fp_due_to_wrong_class" only when iou>=thres and classes don't match. I think this is what you are referring to?

**Comment by nisyad-ms** [comment]
Moved to: https://github.com/microsoft/visionmetrics/pull/57

File: visionmetrics/detection/confusion_matrix.py (hunk 1 above)
Comment (nisyad-ms) [reply visionmetrics/detection/confusion_matrix.py:124]: https://github.com/microsoft/visionmetrics/pull/57

**Pull request closed without merging**